src/modules/launcher/Wox.Plugin/PluginPair.cs

```diff
@@ -167,6 +171,19 @@ private bool CreatePluginInstance()
                 return false;
             }
 
+            // Validate plugin ID to prevent bypassing the GPO by changing the ID in the plugin.json file.
+            string pluginID = (string)type.GetProperty("PluginID", BindingFlags.Public | BindingFlags.Static)?.GetValue(null);
+            if (pluginID == null)
+            {
+                Log.Error($"Can't validate plugin ID of plugin <{Metadata.Name}> in {Metadata.ExecuteFilePath}: The static property <Main.PluginID> was not found.", MethodBase.GetCurrentMethod().DeclaringType);
+                return false;
+            }
+            else if (pluginID != Metadata.ID)
```

Comment:
So the pluginId must be the same that's on the plugin's json file. Wondering where to add this info. Does "doc/thirdPartyRunPlugins.md" make sense?

Reply:
No. The file you mentioned is a list of third-party plugins for end-users. And this here is a dev instruction. Do we have dev doc for plugins?

Reply:
We don't. That's why I suggested we add it there. Not sure, just seems like it should be somewhere? 🤷 But I'm also OK with it not being added.

Reply:
I will update https://github.com/microsoft/PowerToys/blob/main/doc/devdocs/modules/launcher/new-plugin-checklist.md later.

Reply:
Nice find. Something there definitely makes sense. Thank 😄